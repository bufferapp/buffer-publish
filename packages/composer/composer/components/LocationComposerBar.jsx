import React, { useState } from 'react';
import styled from 'styled-components';
import { blue } from '@bufferapp/ui/style/colors';
import Link from '@bufferapp/ui/Link';
import PropTypes from 'prop-types';

import ComposerActionCreators from '../action-creators/ComposerActionCreators';

function Check() {
  return (
    <svg
      style={{ width: '10px', height: '10px' }}
      xmlns="http://www.w3.org/2000/svg"
      ariaHidden="true"
      className="svg-icon"
      viewBox="0 0 512 512"
    >
      <path
        fill="currentColor"
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
      />
    </svg>
  );
}

const Container = styled.div`
  display: inline-flex;
  margin: 5px 0 10px;
  width: 100%;
`;

const LabelWrapper = styled.div`
  font-weight: 600;
  font-size: 12px;
  color: #343e47;
  width: 147px;
  padding: 6px 15px 0 10px;
  box-sizing: border-box;
`;
const InputWrapper = styled.div`
  resize: none;
  padding: 6px 15px 0 10px;
  width: 100%;
  font-size: 12px;
  font-weight: 500;
  box-sizing: border-box;
  input {
    vertical-align: top;
  }
`;

const Checkbox = styled.button`
  width: 15px;
  height: 15px;
  border: ${props =>
    props.checked ? `1px solid ${blue}` : '1px solid #b8b8b8'};
  background: ${props => (props.checked ? blue : '#fff')};
  border-radius: 4px;
  margin-right: 5px;
  padding: 0 1px;
  box-sizing: border-box;
  color: #fff;
`;

const CheckboxLabel = styled.button`
  font-family: inherit;
  font-size: inherit;
  border: 0;
  margin: 0;
  padding: 0;
  outline: none;
  background: none;
`;

const LearnMoreLink = styled(Link)`
  font-size: 12px;
  margin-left: 10px;
`;
function LocationComposerBar({
  draftId,
  isTaggingPageLocation,
  selectedProfiles,
  isInstagram,
}) {
  const shouldShow =
    isInstagram &&
    selectedProfiles.some(profile => profile.instagramDirectEnabled);
  const [checked, setChecked] = useState(!!isTaggingPageLocation);

  const handleClick = () => {
    setChecked(!checked);
    ComposerActionCreators.updateDraftIsTaggingPageLocation(draftId, !checked);
  };

  return shouldShow ? (
    <Container>
      <LabelWrapper>Location</LabelWrapper>
      <InputWrapper>
        <Checkbox
          type="button"
          checked={checked}
          aria-label="Tag with connected Facebook Page location"
          onClick={handleClick}
        >
          <Check />
        </Checkbox>
        <CheckboxLabel type="button" onClick={handleClick}>
          Connected Facebook Page location{' '}
        </CheckboxLabel>
        <LearnMoreLink newTab href="http://buffer.com">
          Learn more
        </LearnMoreLink>
      </InputWrapper>
    </Container>
  ) : null;
}

LocationComposerBar.propTypes = {
  draftId: PropTypes.string.isRequired,
  isTaggingPageLocation: PropTypes.bool.isRequired,
  selectedProfiles: PropTypes.arrayOf(
    PropTypes.shape({ instagramDirectEnabled: PropTypes.bool })
  ).isRequired,
  isInstagram: PropTypes.bool.isRequired,
};

export default LocationComposerBar;
