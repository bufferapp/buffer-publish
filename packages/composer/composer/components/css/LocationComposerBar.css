.locationComposerBar {
    padding: 10px;
    width: 100%;
    box-sizing: border-box;
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
    float: left;
    border-top: 1px solid #E6EBEF;
    margin-top: -1px;
}

.locationComposerBarWithMedia {
    padding: 10px;
    width: 100%;
    box-sizing: border-box;
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
    float: left;
}

.locationFieldContainer {
    float: left;
    padding-top: 1px;
    position: relative;
    display: flex;
    width: 100%;
}

.locationFieldLabel {
    font-weight: 600;
    font-size: 12px;
    color: #343e47;
    width: 110px;
    padding-top: 6px;
}

.locationAutocompleteContainer {
    flex-grow: 1;
}

.locationFieldInput {
    border-radius: 4px;
    padding-left: 8px;
    padding-top: 8px;
    padding-bottom: 8px;
    padding-right: 50px;
    margin-left: 4px;
    border: 1px solid #E6EBEF;
    box-sizing: border-box;
    font-weight: 500;
    font-size: 12px;
    width: 100%;
}

.locationMenu {
    border-radius: 3px;
    box-shadow: rgba(0, 0, 0, 0.1) 0px 2px 12px;
    background: #ffffff;
    overflow: auto;
    position: absolute;
    margin-left: 5px;
    overflow: auto;
    width: 220%;
    max-width: 600px;
    z-index: 100;
    font-size: 12px;
}

.locationFieldRemoveInput {
    background-image: url("data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHZpZXdCb3g9IjAgMCAxNiAxNiIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj4KPGcgaWQ9IkNhbnZhcyIgZmlsbD0ibm9uZSI+CjxnIGlkPSJYIj4KPGNpcmNsZSBpZD0iRWxsaXBzZSIgY3g9IjgiIGN5PSI4IiByPSI4IiBmaWxsPSIjMTkyNTM0Ii8+CjxnIGlkPSJHcm91cCI+CjxyZWN0IGlkPSJSZWN0YW5nbGUgMiIgd2lkdGg9IjEiIGhlaWdodD0iMTAiIHJ4PSIwLjUiIHRyYW5zZm9ybT0ibWF0cml4KDAuNzA3MTA3IC0wLjcwNzEwNyAwLjcwNzEwNyAwLjcwNzEwNyA0IDQuNzA3MTEpIiBmaWxsPSJ3aGl0ZSIvPgo8cmVjdCBpZD0iUmVjdGFuZ2xlIDIuMSIgd2lkdGg9IjEiIGhlaWdodD0iMTAiIHJ4PSIwLjUiIHRyYW5zZm9ybT0ibWF0cml4KC0wLjcwNzEwNyAtMC43MDcxMDcgLTAuNzA3MTA3IDAuNzA3MTA3IDExLjc3ODEgNC43MDcxMSkiIGZpbGw9IndoaXRlIi8+CjwvZz4KPC9nPgo8L2c+Cjwvc3ZnPgo=");
    width: 16px;
    height: 16px;
    position: absolute;
    top: 10px;
    right: 8px;
    z-index: 1;
    cursor: pointer;
}

.optionTitle {
    font-weight: bold;
    padding: 0;
}

.optionTextDescription {
    width: 75%;
    float: left;
    padding-left: 10px;
}

.optionTextDescription p {
    padding: 0;
    margin: 0;
}

.optionImage {
    position: relative;
    float: left;
    margin-top: 5px;
    padding-left: 15px;
    width: 32px;
    height: 32px;
    object-fit: cover;
}

.optionImage:after { 
    position: absolute;
    content: ".";
    display: block;
    color: #E6EBEF;
    background-color: #E6EBEF;
    top: 0;
    bottom: 0;
    right: 0;
    left: 15px;
    text-align: center;
}

.optionRow {
    display: flex;
    align-items: center;
    float: left;
    width: 100%;
    cursor: pointer;
    padding-top: 5px;
    padding-bottom: 7px;
}

.optionRow:nth-of-type(1) {
    padding-top: 8px;
}

.optionRowHighlighted {
    background-color: #ededed;
}
