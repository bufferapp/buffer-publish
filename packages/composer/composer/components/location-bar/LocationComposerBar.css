.locationFieldInput {
    border-radius: 4px;
    padding-left: 8px;
    padding-top: 8px;
    padding-bottom: 8px;
    padding-right: 50px;
    margin-left: 4px;
    border: 1px solid #E6EBEF;
    box-sizing: border-box;
    font-weight: 500;
    font-size: 12px;
    width: 100%;
    outline: none;
}
.locationFieldInput:focus {
    border: 1px solid #2d99ec;
}

.locationFieldRemoveInput {
  background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHZpZXdCb3g9IjAgMCAxNiAxNiIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj4KPGcgaWQ9IkNhbnZhcyIgZmlsbD0ibm9uZSI+CjxnIGlkPSJYIj4KPGNpcmNsZSBpZD0iRWxsaXBzZSIgY3g9IjgiIGN5PSI4IiByPSI4IiBmaWxsPSIjMTkyNTM0Ii8+CjxnIGlkPSJHcm91cCI+CjxyZWN0IGlkPSJSZWN0YW5nbGUgMiIgd2lkdGg9IjEiIGhlaWdodD0iMTAiIHJ4PSIwLjUiIHRyYW5zZm9ybT0ibWF0cml4KDAuNzA3MTA3IC0wLjcwNzEwNyAwLjcwNzEwNyAwLjcwNzEwNyA0IDQuNzA3MTEpIiBmaWxsPSJ3aGl0ZSIvPgo8cmVjdCBpZD0iUmVjdGFuZ2xlIDIuMSIgd2lkdGg9IjEiIGhlaWdodD0iMTAiIHJ4PSIwLjUiIHRyYW5zZm9ybT0ibWF0cml4KC0wLjcwNzEwNyAtMC43MDcxMDcgLTAuNzA3MTA3IDAuNzA3MTA3IDExLjc3ODEgNC43MDcxMSkiIGZpbGw9IndoaXRlIi8+CjwvZz4KPC9nPgo8L2c+Cjwvc3ZnPgo=');
  width: 16px;
  height: 16px;
  position: absolute;
  top: 10px;
  right: 8px;
  z-index: 1;
  cursor: pointer;
}