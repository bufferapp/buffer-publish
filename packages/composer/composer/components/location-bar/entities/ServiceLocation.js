class ServiceLocation {
  constructor(id, name, location) {
    this.id = id;
    this.name = name;
    this.location = location;
  }

  get formattedAddress() {
    if (this.location) {
      const { city, country, state, street, zip } = this.location;
      const formattedCity = city && (state || country) ? `${city},` : city;
      const address = [street, formattedCity, state, zip, country]
        .filter(Boolean)
        .join(' ');

      return address;
    }

    return '';
  }
}

export default ServiceLocation;
