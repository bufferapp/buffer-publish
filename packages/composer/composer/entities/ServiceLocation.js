class ServiceLocation {
  constructor(id, name, pictureUrl, checkins, address) {
    this.id = id;
    this.name = name;

    if (pictureUrl && pictureUrl.source) {
      this.pictureUrl = pictureUrl.source;
    } else {
      this.pictureUrl = pictureUrl;
    }

    this.checkins = checkins;
    this.address = address;
  }

  get formattedAddressWithCheckins() {
    if (this.checkins) {
      const numbersInGroupsOf3 = /\B(?=(\d{3})+(?!\d))/g;
      const formattedNumber = this.checkins.toString().replace(numbersInGroupsOf3, ',');
      const formattedCheckins = `${formattedNumber} people checked in here`;

      if (this.address) {
        return `${this.address} Â· ${formattedCheckins}`;
      }

      return formattedCheckins;
    }

    return '';
  }
}

export default ServiceLocation;
