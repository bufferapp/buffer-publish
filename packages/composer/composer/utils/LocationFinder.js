import API from './API';
import ServiceLocation from '../entities/ServiceLocation';

class LocationFinder {
  static url = 'profiles/get_locations.json';

  static findLocations(profileId, query) {
    const params = {
      profile_id: profileId,
      place: query,
    };

    return API.post(LocationFinder.url, params)
      .catch(() => [])
      .then((response) => {
        const locations = response.locations.data;
        return locations.map((l) =>
          new ServiceLocation(l.id, l.name, l.picture, l.checkins, l.single_line_address)
        );
      });
  }
}
export default LocationFinder;
