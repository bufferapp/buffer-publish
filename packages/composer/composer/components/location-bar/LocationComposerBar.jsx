import React from 'react';
import Autocomplete from 'react-autocomplete';
import debounce from 'lodash.debounce';
import PropTypes from 'prop-types';
import LocationFinder from './utils/LocationFinder';
import styles from './LocationComposerBar.css';
import {
  Container,
  Wrapper,
  Label,
  AutocompleteContainer,
  Menu,
  Row,
  Name,
} from './styles';

import ComposerActionCreators from '../../action-creators/ComposerActionCreators';

const LocationComposerBar = ({
  locationName = '',
  locationId,
  instagramProfileId,
  places = [],
  draftId,
  selectedProfiles,
  isInstagram,
}) => {
  const resetInput = () => {
    ComposerActionCreators.updateDraftLocation(draftId, null, '');
  };

  const onMenuVisibilityChange = isOpen => {
    if (!isOpen && locationId === null && locationName !== '') {
      resetInput();
    }
  };

  const getLocations = debounce(query => {
    LocationFinder.findLocations(instagramProfileId, query)
      .then(results => {
        ComposerActionCreators.updateDraftListPlaces(draftId, results);
      })
      .catch(() => {
        ComposerActionCreators.updateDraftListPlaces(draftId, []);
      });
  }, 450);

  const onChange = (event, value) => {
    ComposerActionCreators.updateDraftLocation(draftId, null, value);
    if (value !== '') {
      getLocations(value);
    }
  };

  const shouldShowLocationBar = () => {
    return (
      isInstagram &&
      selectedProfiles.some(profile => profile.instagramDirectEnabled)
    );
  };

  const saveLocation = newPlace => {
    ComposerActionCreators.updateDraftLocation(
      draftId,
      newPlace.id,
      newPlace.name
    );
  };

  const removeLocation = e => {
    e.preventDefault();
    resetInput();
    ComposerActionCreators.updateDraftListPlaces(draftId, []);
  };

  const isLocationSet = () => locationId !== null;
  return (
    shouldShowLocationBar() && (
      <Container>
        <Wrapper>
          <Label>Location</Label>
          <AutocompleteContainer>
            {isLocationSet() && (
              <span
                className={styles.locationFieldRemoveInput}
                onClick={removeLocation}
              />
            )}
            <Autocomplete
              wrapperStyle={{ width: '100%' }}
              value={locationName}
              items={places}
              inputProps={{
                className: styles.locationFieldInput,
                placeholder: 'Start typing a location...',
              }}
              getItemValue={item => item.name}
              onSelect={(value, place) => {
                saveLocation(place);
              }}
              onChange={onChange}
              onMenuVisibilityChange={onMenuVisibilityChange}
              renderMenu={children => <Menu>{children}</Menu>}
              renderItem={(item, highlighted) => (
                <Row highlighted={highlighted} key={item.id}>
                  <Name>{item.name}</Name>
                  <p>{item.formattedAddress}</p>
                </Row>
              )}
            />
          </AutocompleteContainer>
        </Wrapper>
      </Container>
    )
  );
};

LocationComposerBar.propTypes = {
  draftId: PropTypes.string.isRequired,
  locationName: PropTypes.string,
  locationId: PropTypes.string,
  places: PropTypes.arrayOf({}),
  instagramProfileId: PropTypes.string,
  selectedProfiles: PropTypes.arrayOf(
    PropTypes.shape({
      instagramDirectEnabled: PropTypes.bool,
    })
  ).isRequired,
  isInstagram: PropTypes.bool.isRequired,
};

export default LocationComposerBar;
