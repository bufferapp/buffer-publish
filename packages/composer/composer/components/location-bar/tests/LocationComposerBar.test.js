/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/first */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Autocomplete from 'react-autocomplete';
import debounce from 'lodash.debounce';

import ServiceLocation from '../entities/ServiceLocation';
import LocationComposerBar from '../LocationComposerBar';

jest.mock('../../../action-creators/ComposerActionCreators');
jest.mock('../utils/LocationFinder');
jest.mock('lodash.debounce', () => jest.fn(fn => fn));

import ComposerActionCreators from '../../../action-creators/ComposerActionCreators';
import LocationFinder from '../utils/LocationFinder';

const showLocationBar = {
  selectedProfiles: [{ instagramDirectEnabled: true }],
  isInstagram: true,
  hasVideo: true,
  withMediaAttachment: false,
};

describe('Whole component', () => {
  it('renders correctly', () => {
    const sl = new ServiceLocation(
      '3123',
      'Los Angeles',
      'https://example.com/image.jpeg',
      123,
      'LA, USA'
    );

    const tree = renderer
      .create(
        <LocationComposerBar
          {...showLocationBar}
          draftId={'123'}
          locationName={'New York'}
          locationId={'123123'}
          instagramProfileId={'333'}
          places={[sl]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with location name when set', () => {
    const tree = renderer
      .create(
        <LocationComposerBar
          {...showLocationBar}
          draftId={'123'}
          locationName={'New York'}
          instagramProfileId={'333'}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with cross to remove location when set', () => {
    const tree = renderer
      .create(
        <LocationComposerBar
          {...showLocationBar}
          draftId={'123'}
          locationName={'New York'}
          locationId={'123123'}
          instagramProfileId={'333'}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const setup = propOverrides => {
  const props = Object.assign(
    {
      draftId: '1231',
      locationName: '',
      locationId: null,
      instagramProfileId: null,
      places: [],
    },
    propOverrides
  );

  const wrapper = shallow(
    <LocationComposerBar {...props} {...showLocationBar} />
  );

  return {
    props,
    wrapper,
  };
};

describe('Cross remove button', () => {
  it('is shown when locationId is set', () => {
    const { wrapper } = setup({ locationId: '123123' });
    expect(wrapper.find('.locationFieldRemoveInput').length).toBe(1);
  });

  it('is hidden when locationId is not set', () => {
    const { wrapper } = setup({ locationId: null });
    expect(wrapper.find('.locationFieldRemoveInput').length).toBe(0);
  });

  it('resets input content when clicked on', () => {
    ComposerActionCreators.updateDraftLocation.mockReturnValue(true);
    const { wrapper } = setup({ locationId: '123', draftId: '1234' });

    wrapper
      .find('span.locationFieldRemoveInput')
      .simulate('click', { preventDefault() {} });

    expect(ComposerActionCreators.updateDraftLocation).toHaveBeenCalledWith(
      '1234',
      null,
      ''
    );
  });
});

describe('Autocomplete element', () => {
  it('is present in component', () => {
    const { wrapper } = setup();
    expect(wrapper.find(Autocomplete).length).toBe(1);
  });

  it('has correct properties', () => {
    const sl = new ServiceLocation(
      '3123',
      'Los Angeles',
      'https://example.com/image.jpeg',
      123,
      'LA, USA'
    );
    const { wrapper } = setup({
      locationName: 'Los Angeles Test',
      places: [sl],
    });
    const autocompleteProps = wrapper.find(Autocomplete).props();

    expect(autocompleteProps.items).toEqual([sl]);
    expect(autocompleteProps.value).toEqual('Los Angeles Test');
    expect(autocompleteProps.getItemValue(sl)).toEqual('Los Angeles');
  });
});

describe('Autocomplete actions', () => {
  beforeEach(() => {
    ComposerActionCreators.updateDraftLocation.mockReturnValue(true);
  });
  afterEach(() => jest.resetAllMocks());

  it('updates location list when changed', async () => {
    LocationFinder.findLocations.mockReturnValue(Promise.resolve(['t']));
    const { wrapper } = setup({ draftId: '1234', instagramProfileId: '676' });
    const autocompleteProps = wrapper.find(Autocomplete).props();
    await autocompleteProps.onChange(null, 'LA Test');

    expect(ComposerActionCreators.updateDraftLocation).toHaveBeenCalledWith(
      '1234',
      null,
      'LA Test'
    );

    expect(debounce).toHaveBeenCalled();
    expect(LocationFinder.findLocations).toHaveBeenCalledWith('676', 'LA Test');
    expect(ComposerActionCreators.updateDraftListPlaces).toHaveBeenCalledWith(
      '1234',
      ['t']
    );
  });

  it('updates draft location when changed', () => {
    const { wrapper } = setup({ draftId: '1234', instagramProfileId: '676' });
    const autocompleteProps = wrapper.find(Autocomplete).props();
    const place = new ServiceLocation(
      '3123',
      'Los Angeles',
      'https://example.com/image.jpeg',
      123,
      'LA, USA'
    );

    autocompleteProps.onSelect(null, place);

    expect(ComposerActionCreators.updateDraftLocation).toHaveBeenCalledWith(
      '1234',
      place.id,
      place.name
    );
  });

  it('resets input content when no location is selected', () => {
    const { wrapper } = setup({
      locationId: null,
      locationName: 'NYC',
      draftId: '1234',
    });
    const autocompleteProps = wrapper.find(Autocomplete).props();

    autocompleteProps.onMenuVisibilityChange(false);

    expect(ComposerActionCreators.updateDraftLocation).toHaveBeenCalledWith(
      '1234',
      null,
      ''
    );
  });
});
