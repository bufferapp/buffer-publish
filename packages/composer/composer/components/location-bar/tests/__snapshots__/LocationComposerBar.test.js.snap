// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Whole component renders correctly 1`] = `
.c0 {
  padding: 3px 10px 11px;
  width: 100%;
  box-sizing: border-box;
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
  float: left;
}

.c1 {
  float: left;
  padding-top: 1px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c2 {
  font-weight: 600;
  font-size: 12px;
  color: #3D3D3D;
  width: 120px;
  padding-top: 6px;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Location
    </label>
    <span
      className="c3"
    >
      <span
        className="locationFieldRemoveInput"
        onClick={[Function]}
      />
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          aria-autocomplete="list"
          aria-expanded={false}
          autoComplete="off"
          className="locationFieldInput"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Start typing a location..."
          role="combobox"
          value="New York"
        />
      </div>
    </span>
  </div>
</div>
`;

exports[`Whole component renders with cross to remove location when set 1`] = `
.c0 {
  padding: 3px 10px 11px;
  width: 100%;
  box-sizing: border-box;
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
  float: left;
}

.c1 {
  float: left;
  padding-top: 1px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c2 {
  font-weight: 600;
  font-size: 12px;
  color: #3D3D3D;
  width: 120px;
  padding-top: 6px;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Location
    </label>
    <span
      className="c3"
    >
      <span
        className="locationFieldRemoveInput"
        onClick={[Function]}
      />
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          aria-autocomplete="list"
          aria-expanded={false}
          autoComplete="off"
          className="locationFieldInput"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Start typing a location..."
          role="combobox"
          value="New York"
        />
      </div>
    </span>
  </div>
</div>
`;

exports[`Whole component renders with location name when set 1`] = `
.c0 {
  padding: 3px 10px 11px;
  width: 100%;
  box-sizing: border-box;
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
  float: left;
}

.c1 {
  float: left;
  padding-top: 1px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c2 {
  font-weight: 600;
  font-size: 12px;
  color: #3D3D3D;
  width: 120px;
  padding-top: 6px;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Location
    </label>
    <span
      className="c3"
    >
      <span
        className="locationFieldRemoveInput"
        onClick={[Function]}
      />
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          aria-autocomplete="list"
          aria-expanded={false}
          autoComplete="off"
          className="locationFieldInput"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Start typing a location..."
          role="combobox"
          value="New York"
        />
      </div>
    </span>
  </div>
</div>
`;
